sdk/modelsrepository/azure-iot-modelsrepository/src/main/java/com/azure/iot/modelsrepository/ModelsDependencyResolution.java

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.iot.modelsrepository;
+
+/**
+ * The model dependency resolution options.
+ */
+public enum ModelsDependencyResolution {
+
+    /**
+     * Disable model dependency resolution.
+     */
+    DISABLED,
+
+    /**
+     * Enable model dependency resolution. The client will parse models and calculate dependencies recursively.
+     */
+    ENABLED,
+
+    /**
+     * Try to get pre-computed model dependencies using .expanded.json.
+     * If the model expanded form does not exist fall back to DependencyResolutionOption.Enabled processing.
+     */
+    TRY_FROM_EXPANDED,
```

Comment:
something like AUTO_COMPUTE_DEPENDENCY may sound better?

Reply:
or just AUTO_COMPUTE

Reply:
While I don't disagree with you, the .NET version also has the same enum values, we will shoot for consistency here.

Reply:
The try from expanded concept will likely look different in a future revision of the service. But yea lets stick with the same terminology for consistency in the first preview.